content/blog/2024-05-22-rustl-2024-unconference.md

```diff
@@ -0,0 +1,143 @@
+- Text layout and edition.
+- Winit adoption.
+- Accessibility.
+- Potential for further collaboration.
+- Requests for new Rust features.
+
+Note that I'm skipping a lot topics on the agenda, and other side discussions.
+It feels like those topics were the most prominent during those two days.
+
+There was also a fair amount of informal chatter around LogLog Games' incendiary [Leaving Rust gamedev after 3 years](https://loglog.games/blog/leaving-rust-gamedev/) article.
+I think those discussions mostly reflected the Bevy community's consensus, that the article had some flaws but overall pointed at very clear weaknesses in the Rust GUI / game engine ecosystem.
+
+### Build tooling
+
+Everyone agreed that a better solution was needed to ship non-TUI Rust apps.
+
+There was discussion about various platforms and build system, and the difficulty of having to eg write Java code for an Android port, and whether anyone wanted to take responsibility for maintaining that glue layer for the rest of the ecosystem.
+
+Unfortunately the discussions on this subject didn't really go anywhere, as far as I could tell.
+Packaging Rust apps is still very much an open problem.
+
+### Text layout and edition
+
+There was some widespread interest in the Linebender ecosystem's text handling crates.
+From what I remember, Rik Arends and others had concerns about whether eg Parley and Swash added dependencies to our Vello renderer (they don't).
+
+Some people also wanted Parley to be decoupled from Swash.
+
+People we especially interested in the idea of having access to a common text-editing widget, or at least infrastructure to create one.
+That widget would handle IME, text selection, accessibility actions, etc, in a way that would feel native to different platforms with their own text-editing quirks (mobile, MacOS, Linux with vim mode, etc).
+
+### Winit
+
+By now all of the Rust ecosystem has firmly converged on Winit.
+
+(Well, not all! [One small project](https://github.com/makepad/makepad) with indomitable maintainers still holds out against the invaders.)
+
+The consensus during this discussion seemed to be that Winit was there to stay, and that any changes we wanted would probably come in the form of upstream contributions, not competing projects.
+
+Glazier is still shelved and we're planning to port its features to Winit until it reaches feature parity.
+
+A discussion that came up was the idea of splitting more "vocabulary types" from Winit into standalone crates.
+Vocabulary crates, eg crates that only define types and very little code, have a very low cost in memory and build types, while allowing projects to interop with eg Winit without actually adding a dependency on Winit.
+
+This was connected with the discussions about text edition: having a vocabulary crate that would list Winit's events and also Parley's cursor movements would allow for a very low-dependency text edition widget.
```

Comment:
Text edition?

Reply:
Text-editing, as in a content-editable textbox.

Reply:
We use 'edition' as the passive form of 'editing' at work, but I don't know if it's actual English. I work for a French company, so it might be a frenchism.

Reply:
Yeah, that's a Frenchism (as a native English speaker with fluency in French) :)

Reply:
Huh. It's been a while since I've been caught doing one of those. Usually it's the other way around these days.